src/browser-connection/set.js

```diff
@@ -0,0 +1,178 @@
+import { EventEmitter } from 'events';
+import { Promise } from 'es6-promise';
+import flatten from 'flatten';
+import LocalBrowserConnection from './local';
+import { MESSAGE, getText } from '../messages';
+import remove from '../utils/array-remove';
+
+
+function getConst (name, defaultValue) {
+    var value = GLOBAL[name];
+
+    if (value === void 0)
+        return defaultValue;
+
+    return value;
+}
+
+export default class BrowserConnectionSet extends EventEmitter {
+    constructor () {
+        super();
+
+        this.BROWSER_CONNECTION_READY_TIMEOUT = getConst('BROWSER_CONNECTION_READY_TIMEOUT', 30 * 1000);
+        this.WAITING_FOR_IDLE_TIMEOUT         = getConst('WAITING_FOR_IDLE_TIMEOUT', 10000);
+
+        this.connections        = [];
+        this.bindedErrorHandler = this._connectionErrorHandler.bind(this);
+        this.promisedCloses     = [];
+        this.task               = null;
+
+        //NOTE: We're setting an empty error handler, because Node kills the process on an 'error' event
+        //if there is no handler. See https://nodejs.org/api/events.html#events_class_events_eventemitter
+        this.on('error', () => {});
+    }
+
+    _cleanJobQueue (bc) {
+        if (bc instanceof LocalBrowserConnection)
+            bc.jobQueue = [];
+        else {
+            if (!bc.jobQueue.length || !this.task)
+                return;
+
+            if (this.task.pendingBrowserJobs.indexOf(bc.jobQueue[0]) > -1)
+                bc.removeJob(bc.jobQueue[0]);
+        }
+    }
+
+    _createConnectionIdlePromise (bc) {
+        return new Promise(resolve => {
+            if (bc.idle || bc.closed) {
+                resolve();
+                return;
+            }
+            var waitingForIdleTimeout = setTimeout(() => resolve(), this.WAITING_FOR_IDLE_TIMEOUT);
+
+            var eventHandler = () => {
+                clearTimeout(waitingForIdleTimeout);
+                resolve();
+            };
+
+            bc.once('idle', eventHandler);
+            bc.once('closed', eventHandler);
```

Comment:
What will happen if timeout will be triggered, then `idle` will be triggered, then `closed` will be triggered? Handler will be called 3 times. Use https://github.com/isaacs/once